lib/ansible/plugins/strategy/__init__.py

```diff
@@ -844,8 +844,13 @@ def run_handlers(self, iterator, play_context):
             for handler in handler_block.block:
                 if handler.notified_hosts:
                     handler_vars = self._variable_manager.get_vars(play=iterator._play, task=handler)
-                    handler_name = handler.get_name()
-                    result = self._do_handler_run(handler, handler_name, iterator=iterator, play_context=play_context)
+                    if C.HANDLERS_TAGS_SUPPORT:
+                        if not ((handler.action == 'include' and handler.evaluate_tags([], play_context.skip_tags, all_vars=handler_vars))
```

Comment:
should we also treat include_tasks and include_role special?

Reply:
Wouldn't be the behavior different from the 'normal' tasks then? https://github.com/ansible/ansible/blob/445ff39/lib/ansible/playbook/block.py#L376

Reply:
i forgot we refactored that, the include_ logic is now elsewhere .. but i don't know if it covers handlers (@sivel, do you know of top of your head?)

Reply:
Forgout about this... So if I am reading the code correctly, that logic is now in strategies, e.g. linear: https://github.com/ansible/ansible/blob/b01b1d40cfe0831b11095bac5af3eb7b8bc49ece/lib/ansible/plugins/strategy/linear.py#L364

We do not do that for handlers: https://github.com/ansible/ansible/blob/264945d75fad4c11bde1a6d236fda0fce909c335/lib/ansible/plugins/strategy/__init__.py#L922

So we should add that there too.

Reply:
Hrm, so this is not a correct place to do this as we don't have host vars yet. This should be moved to `_do_handler_run()`. Which also solves filtering tags on included tasks.